src/lib.rs

```diff
@@ -210,47 +210,88 @@ impl StackPage {
     }
 }
 
+#[cfg(feature = "psci")]
+#[repr(C, align(16))] // align to the aarch64 stack requirements
+struct StartCoreStack<F> {
+    trampoline_ptr: unsafe extern "C" fn(*mut StartCoreStack<F>) -> !,
```

Comment:
Can we make this function safe, and take a `&mut` rather than `*mut`? Then the implicit requirement is that the caller must follow Rust's aliasing rules for the pointer it passes, which as far as I can see it does.

Reply:
Fair, we can - this comes from my initial implementation which used quite a bit more pointer trickery, but in the current from it's indeed not needed.